### PR TITLE
move CI from travis to github actions

### PR DESCRIPTION
**Pull request by gidsg**
https://trello.com/c/DjnmsKpt/623-1-migrate-supplier-frontend-ci-from-travis-to-github-actions

# Notes
* We are adding some ubuntu dependencies, this is to workaround an issues with lxml when using python 3.9 (I was getting the issue described in https://stackoverflow.com/questions/15759150/src-lxml-etree-defs-h931-fatal-error-libxml-xmlversion-h-no-such-file-or-di only when using Python 3.9)
* We were using coveralls but this relied on a built-in integration with Travis CI so it has been removed in this PR

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    - name: Setup Python cache
+      uses: actions/cache@v2
+      with:
+        path: venv
+        key: venv-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('**/requirements-dev.txt') }}
+
+    - name: Run python tests
+      env: 
+        PYTEST_ARGS: --cov=app --cov-report=term-missing
```
Comment (bjgill) [inline .github/workflows/test.yml:44]: If we're not using coveralls, do we need this?

File: requirements-dev.in [hunk 2]
```diff
@@ -3,7 +3,6 @@
 pip-tools>=5.4.0
 
 coverage==4.5.3
```
Comment (lfdebrux) [inline requirements-dev.in:5]: We can probably get rid of coverage altogether

**Review by bjgill — APPROVED** [review]
Nice. One question, but not blocking.

Also, do you have a link to the issue with Python 3.9?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Read .nvmrc
+      run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+      id: nvm
```
Comment (lfdebrux) [inline .github/workflows/test.yml:14]: Is this really the best way to read the nvmrc file? 😱 

Something to look at later I suspect

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      uses: actions/setup-node@v1
+      with:
+        node-version: '${{ steps.nvm.outputs.NVMRC }}'
+
+    - uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
```
Comment (lfdebrux) [inline .github/workflows/test.yml:29]: Why does restore keys have a hanging `-`? Excuse my ignorance

File: .github/workflows/test.yml (hunk 3 above)
Comment (gidsg) [reply .github/workflows/test.yml:14]: Good point, meant to add clarification around this. There are a number of approaches to replicate what we were doing before with nvm on travis CI, this one I got from https://github.com/actions/setup-node/issues/32#issuecomment-539794249

**Comment by lfdebrux** [comment]
We'll have to remove Travis CI from the list of required checks for this branch to be able to merge this PR

File: .github/workflows/test.yml (hunk 3 above)
Comment (lfdebrux) [reply .github/workflows/test.yml:14]: Awesome, clarification would be good

File: .github/workflows/test.yml (hunk 4 above)
Comment (gidsg) [reply .github/workflows/test.yml:29]: 🤷 this is some magic I derived from https://github.com/actions/setup-node/issues/32#issuecomment-539794249 I suspect is just a clearer way of denoting that prefix.

**Review by lfdebrux — APPROVED** [review]
Looking good! ✅

File: .github/workflows/test.yml (hunk 3 above)
Comment (gidsg) [reply .github/workflows/test.yml:14]: @lfdebrux Scratch that it appears from later comments nvm got bundled so there may be a nicer approach closer to what we do in the makefile I will take a look.

File: .github/workflows/test.yml (hunk 1 above)
Comment (gidsg) [reply .github/workflows/test.yml:44]: I wasn't sure if some other tooling used this but if not, I'm happy to remove both this and `coverage`

File: .github/workflows/test.yml (hunk 3 above)
Comment (gidsg) [reply .github/workflows/test.yml:14]: @lfdebrux thanks for raising this and making me look again at it. I've changed it now to use `nvm` directly in ab5307c.

**Review by lfdebrux — APPROVED** [review]

File: .github/workflows/test.yml (hunk 3 above)
Comment (lfdebrux) [reply .github/workflows/test.yml:14]: Cool!

File: requirements-dev.in (hunk 2 above)
Comment (gidsg) [reply requirements-dev.in]: removed in 919a4c3

File: .github/workflows/test.yml (hunk 1 above)
Comment (gidsg) [reply .github/workflows/test.yml]: removed in 919a4c3

**Pull request merged**